### PR TITLE
Text cell with help text in grid

### PR DESCRIPTION
**Pull request by lassopicasso**
## Description

Two new configuration options have been introduced for grid cells:
1. In addition to the `text` prop, the current text cell now also supports the `help` prop, which allows to include help text along with the `text` prop. 
2. A new type of cell has been added, which functions similarly to the text cell, but instead refers to text resources from a component. It uses the property `labelFrom` with the corresponding component ID. In the referenced component, you can access the `title` (which serves as the label for the component), `description`, and `help` to provide help text.

Example 1:
```
            "cells": [
              { "text": "laan-bolig", "help": "Dette er en hjelpetekst til laan-bolig" },
              ...
            ]
```

Example 2:
```
            "cells": [
              { "labelFrom": "<component-id>"},
              ...
            ]
```

## Related Issue(s)

- closes #https://github.com/Altinn/app-frontend-react/issues/1230

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [x] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [x] Has been added/updated
https://github.com/Altinn/altinn-studio-docs/pull/1011
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Changes/additions to component properties
  - [x] Changes are reflected in both `src/layout/layout.d.ts` and `layout.schema.v1.json`, and these are all backwards-compatible
  - [ ] No changes made
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [ ] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Comment by olemartinorg** [comment]
Hmm. A few comments on this one:
1. I don't think we need the logic for `aria-describedby`. Components in Grid are described by their own text resources, as they are required for [mobile support](https://docs.altinn.studio/app/development/ux/components/grid/#mobile-support). Meaning, the components in the table have everything they need for accessibility already. Also the description should come from the `description` text resource binding - help text is something else.
2. We can't make the assumption that a text cell describes all components in the same row. It's entirely valid to have a row that is `[text cell, component, text cell, component]`, and so making assumptions that the first text cell describes the last component is most likely false.
3. The thing I remember we discussed was to avoid duplicating the help text by referencing a component (the other way around from what is done here). Meaning that you could for example implement a row like:

```json
{
  "cells": [
    { "textFrom": "myComponent" },
    { "component": "myComponent" }
  ]
}
```

And such a configuration would fetch the texts (i.e. `title`, `help` and possibly also `description`) from the target component and put them in the text cell. That way you get most of the same effects of the `aria-describedby` feature added here, but also:

- No need to add the text resource bindings twice (both on the component and in the text cell)
- We can implement functionality to make the title/description/label to work as a label (i.e. clicking on it focuses the target field)

**Comment by lassopicasso** [comment]
Understood, thanks for the feedback @olemartinorg ! 👍🙏

File: src/layout/Grid/GridComponent.tsx [hunk 1]
```diff
@@ -76,29 +82,48 @@ export function GridRowRenderer({ row, isNested, mutableColumnSettings }: GridRo
           mutableColumnSettings[cellIdx] = cell.columnOptions;
         }
 
-        if (cell && 'text' in cell && cell.text) {
+        const closestComponent = () => {
+          if (cell && 'labelFrom' in cell && cell?.labelFrom) {
+            const idToFind = cell.labelFrom;
+            return node.flat(true).find((n) => n.item.id === idToFind || n.item.baseComponentId === idToFind);
+          }
+        };
```
Comment (olemartinorg) [inline src/layout/Grid/GridComponent.tsx:90]: Nitpicking here, but this code could probably just be moved down into the other `if` below, and it has no need to be a function, so you could just pass the found node to `<CellWithLabel>` directly.

File: src/layout/Grid/GridComponent.tsx [hunk 2]
```diff
@@ -131,6 +156,8 @@ function InternalRow({ header, readOnly, children }: InternalRowProps) {
 interface CellProps {
   className?: string;
   columnStyleOptions?: ITableColumnProperties;
+  referenceComponent?: LayoutNode;
+  help?: string;
 }
```
Comment (olemartinorg) [inline src/layout/Grid/GridComponent.tsx:161]: Could probably just make another type for `<CellWithLabel>`. Mixing these two cell types could just be confusing.

**Review by olemartinorg — APPROVED** [review]
This looks great! 🚀 Some very minor and optional nitpicks in the comments.

**Pull request merged**